pkg/pgmodel/querier/querier.go

```diff
@@ -274,64 +286,86 @@ func (q *pgxQuerier) queryMultipleMetrics(filter metricTimeRangeFilter, cases []
 
 // getMetricTableName gets the table name for a specific metric from internal
 // cache. If not found, fetches it from the database and updates the cache.
-func (q *pgxQuerier) getMetricTableName(metric string) (string, error) {
-	var err error
-	var tableName string
+func (q *pgxQuerier) getMetricTableName(schema, metric string) (string, string, string, error) {
+	var (
+		err             error
+		tableName       string
+		tableSchema     string
+		seriesTableName string
+	)
 
-	tableName, err = q.metricTableNames.Get(metric)
+	tableSchema, tableName, seriesTableName, err = q.metricTableNames.Get(schema, metric)
 
 	if err == nil {
-		return tableName, nil
+		return tableSchema, tableName, seriesTableName, nil
 	}
 
 	if err != errors.ErrEntryNotFound {
-		return "", err
+		return "", "", "", err
 	}
 
-	tableName, err = q.queryMetricTableName(metric)
+	tableSchema, tableName, seriesTableName, err = q.queryMetricTableName(schema, metric)
 
 	if err != nil {
-		return "", err
+		return "", "", "", err
+	}
+
+	// In case of unknown schema, we need to set it as a default one.
+	if schema == "" {
+		err = q.metricTableNames.SetDefaultSchema(tableSchema, metric, tableName, seriesTableName)
+
+		if err != nil {
+			return "", "", "", err
+		}
 	}
 
-	err = q.metricTableNames.Set(metric, tableName)
+	err = q.metricTableNames.Set(tableSchema, metric, tableName, seriesTableName)
 
-	return tableName, err
+	return tableSchema, tableName, seriesTableName, err
 }
 
-func (q *pgxQuerier) queryMetricTableName(metric string) (string, error) {
+func (q *pgxQuerier) queryMetricTableName(schema, metric string) (string, string, string, error) {
 	res, err := q.conn.Query(
 		context.Background(),
 		getMetricsTableSQL,
 		metric,
 	)
 
 	if err != nil {
-		return "", err
+		return "", "", "", err
 	}
 
-	var tableName string
+	var (
+		tableName       string
+		tableSchema     string
+		seriesTableName string
+	)
 	defer res.Close()
-	if !res.Next() {
-		return "", errors.ErrMissingTableName
-	}
+	for res.Next() {
+		if err := res.Scan(&tableName, &tableSchema, &seriesTableName); err != nil {
+			return "", "", "", err
+		}
 
-	if err := res.Scan(&tableName); err != nil {
-		return "", err
+		if schema == "" || schema == tableSchema {
```

Comment:
This logic means if we don't have a schema, this will return the first row. Didn't we want the schema to be prom_data if not specified? Also, do we want this logic in Go or in SQL?

Reply:
Right, I might have mixed it up with getting whatever metric was first created. Will change that. Let me see if I can move this logic to SQL so we only get zero or one rows returned.